d2/src/main/java/com/linkedin/d2/balancer/util/downstreams/FSBasedDownstreamServicesFetcher.java

```diff
@@ -31,17 +31,19 @@ public class FSBasedDownstreamServicesFetcher implements DownstreamServicesFetch
 {
   private final String _d2FsPath;
   private final String _d2ServicePath;
+  private final String _fsFileExtension;
 
-  public FSBasedDownstreamServicesFetcher(String d2FsPath, String d2ServicePath)
+  public FSBasedDownstreamServicesFetcher(String d2FsPath, String d2ServicePath, String fsFileExtension)
```

Comment:
can you also make this backward compatible? Then you don't need to change some files in this PR?

Reply:
Done